app/livechat/server/api/v1/contact.js

```diff
@@ -0,0 +1,37 @@
+import { Match, check } from 'meteor/check';
+
+import { API } from '../../../../api/server';
+import { Contacts } from '../../lib/Contacts';
+
+
+API.v1.addRoute('contact', { authRequired: true }, {
+	post() {
+		try {
+			check(this.bodyParams, {
+				_id: Match.Maybe(String),
+				name: String,
+				email: Match.Maybe(String),
+				phone: Match.Maybe(String),
+			});
+
+			const contactParams = this.bodyParams;
+			if (this.bodyParams.phone) {
+				contactParams.phone = { number: this.bodyParams.phone };
+			}
+			const contact = Contacts.saveContact(contactParams);
```

Comment:
Why not using `Livechat.registerGuest` method?

Reply:
registerGuest is based on token, we don't use token here.
I've created the "Contacts" service level, so he is using `LivechatVisitors.insert` method.

Reply:
You can generate a `new token` and pass it as a parameter.